spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/AuthParameterUtils.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.autoconfigure;
+
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.pulsar.common.util.ObjectMapperFactory;
+
+/**
+ * Utility methods for Pulsar authentication parameters.
+ *
+ * @author Alexander Preuß
+ */
+public final class AuthParameterUtils {
+
+	private static final Pattern KEBAB_CASE_PATTERN = Pattern.compile("-(.)");
+
+	private AuthParameterUtils() {
+
+	}
+
+	private static String convertKebabCaseToCamelCase(String kebabString) {
+		return KEBAB_CASE_PATTERN.matcher(kebabString).replaceAll(mr -> mr.group(1).toUpperCase());
+	}
+
+	private static Map<String, String> convertWellKnownLowerCaseKeysToCamelCase(Map<String, String> params) {
+		return params.entrySet().stream().collect(Collectors
+				.toMap(entry -> WellKnownAuthParameters.getCamelCaseKey(entry.getKey()), Map.Entry::getValue));
+	}
+
+	private static Map<String, String> convertKebabCaseKeysToCamelCase(Map<String, String> params) {
+		return params.entrySet().stream()
+				.collect(Collectors.toMap(entry -> convertKebabCaseToCamelCase(entry.getKey()), Map.Entry::getValue));
+	}
+
+	protected static String maybeConvertToEncodedParamString(Map<String, String> params) {
+		if (params == null) {
+			return null;
+		}
+		params = convertWellKnownLowerCaseKeysToCamelCase(params);
```

Comment:
It may be helpful to add a comment above this about the fact that env vars will come in as lower-case as we recommend mapping the keys as env vars as a single word w/no underscores and this is how we "reconstitute" them to be in their proper camel-case form.